quickwit-index-config/src/tag_pruning.rs

```diff
@@ -0,0 +1,478 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::BTreeSet;
+use std::fmt::Display;
+
+use tantivy::query::QueryParserError as TantivyQueryParserError;
+use tantivy_query_grammar::{Occur, UserInputAst, UserInputLeaf, UserInputLiteral};
+
+use crate::QueryParserError;
+
+fn user_input_ast_to_tags_filter_ast(user_input_ast: UserInputAst) -> Option<TagFilterAst> {
+    let filters_ast = collect_tag_filters(user_input_ast);
+    let term_filters_ast = simplify_ast(filters_ast)?;
+    Some(expand_to_tag_ast(term_filters_ast))
+}
+
+/// Returns true if and only if tag is of form `{field_name}:any_value`.
+pub fn match_tag_field_name(field_name: &str, tag: &str) -> bool {
+    tag.len() > field_name.len()
+        && tag.as_bytes()[field_name.len()] == b':'
```

Comment:
My guess is we only support ASCII.

Reply:
The code actually works with utf8. It's a fun property of utf8. 
Simple ascii codes stay unchanged.

(also this is not used here by simple ascii code cannot appear as part as a larger codepoint.)

Reply:
(your concern was very good question though)